pkg/pgmodel/model/pgsafetype/custom_pgtypes.go

```diff
@@ -0,0 +1,99 @@
+package pgsafetype
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jackc/pgtype"
+	"github.com/timescale/promscale/pkg/pgmodel/model"
+)
+
+// Text is a custom pgtype that wraps the pgtype.Text. It is safe from null characters.
+type Text struct {
+	pgtype.Text
+}
+
+func (t *Text) Set(src interface{}) error {
+	text := src.(string)
+	if err := t.Text.Set(sanitizeNullChars(text)); err != nil {
+		return fmt.Errorf("safe-text: %w", err)
+	}
+	return nil
+}
+
+func (t *Text) Get() interface{} {
+	return revertSanitization(t.Text.Get().(string))
+}
+
+func (t *Text) AssignTo(dst interface{}) error {
+	if err := t.Text.AssignTo(dst); err != nil {
+		return fmt.Errorf("safe-text: %w", err)
+	}
+	return nil
+}
+
+func (t *Text) Scan(src interface{}) error {
+	if err := t.Text.Scan(src); err != nil {
+		return fmt.Errorf("safe-text: %w", err)
+	}
+	return nil
+}
+
+// TextArray is a custom pgtype that wraps the pgtype.TextArray. It is safe from null characters.
+type TextArray struct {
+	pgtype.TextArray
+}
+
+func (t *TextArray) Set(src interface{}) error {
+	textArray := src.([]string)
+	for i := range textArray {
+		textArray[i] = sanitizeNullChars(textArray[i])
+	}
```

Comment:
It's probably more efficient to build a pgtype.Text slice here. You should check the work done inside pgtype.TextArray.Set() my hunch would be it does less work when passed a []pgtype.Text

Reply:
I will write some go benchmarks and test this along with what is said at https://github.com/timescale/promscale/pull/530#discussion_r595659173

Reply:
Benchmark results

```
// using []pgtype.Text in pgsafetype.TextArray.Set(): BenchmarkTypeArray-8   	   85900	     13161 ns/op
// using []string in pgsafetype.TextArray.Set(): 	  BenchmarkTypeArray-8   	   84028	     16224 ns/op
```
clearly, []pgtype.Text is the better one (as you said) as the results were consistent in repeated benchmarks.

Reply:
Updating to pgtype.Text way